### ISSUE TITLE
Great job!

### ISSUE THREAD

**Issue by dbousamra** [rbaron/awesomenes#1]
Hi

I'm not a Lobster's member :( Github needs to allow comments. I've written a NES emulator (http://github.com/dbousamra/hnes), so I know how tricky the NES is. I found your code really clean, and easy to understand. Nice work!

**Comment by rbaron** [comment]
Hey @dbousamra. Thanks for the message! Indeed it's been really an adventure writing this. The funny thing is I considered doing this in Haskell! I did the advent of code challenge in Haskell last year and had a lot of fun. In the end I wans't as brave as you, though. Really nice work there too!

**Comment by rbaron** [comment]
Amazing. Thanks for the tip @jairodasilva. I'll take a closer look at gometalinter!

About the code, when I began writing this, I didin't have a very good mental model of all the moving parts of the NES. What helped me a lot was reading the source code for several other open source emulators out there. In particular, I messed around a lot with [AndreaOrru/LaiNES](https://github.com/AndreaOrru/LaiNES) and [fogleman/nes](https://github.com/fogleman/nes). The first one is amazing in that it is super concise and the info on [nesdev wiki](http://wiki.nesdev.com/w/index.php/Nesdev_Wiki) is often straightforwardly distilled in the code. The second one is an extremely well written and polished NES emulator in Go. I highly recommend taking a look at both!
